### PR TITLE
Add basic auth filter new

### PR DESCRIPTION
**Pull request by crasu**
Just rebased this again to update to the current rolling version.

This is just a kind heads up to let people know there is some option to run ai on the edge with basic authentification. You may close this pr without comment. 

Btw: Running this now for several month without issues.

### CODE REVIEW

**Comment by nima-1102** [comment]
+1 I would also prefer to add at least basic authentication.

File: code/platformio.ini [hunk 1]
```diff
@@ -57,7 +57,6 @@ build_flags =
 	${flags:runtime.build_flags}
     ; ### Sofware options : (can be set in defines.h)
 	-D ENABLE_MQTT 
-    -D ENABLE_INFLUXDB
```
Comment (ThomasCr) [inline code/platformio.ini:60]: was this needed?

File: code/components/jomjol_wlan/read_wlanini.cpp [hunk 2]
```diff
@@ -157,8 +180,8 @@ int LoadWlanFromFile(std::string fn)
             #endif
         }
 
-        /* read next line */
-        if (fgets(zw, sizeof(zw), pFile) == NULL) {
+        if (fgets(zw, 1024, pFile) == NULL)
```
Comment (ThomasCr) [inline code/components/jomjol_wlan/read_wlanini.cpp:183]: why not sizeof(zw) - sounds better for me....

Comment (EmmanuelP) [reply code/components/jomjol_wlan/read_wlanini.cpp:183]: Especially with `char zw[256];` above

**Comment by caco3** [comment]
@crasu I rebased your PR on the latest main, but I am unable to build it. Any clou? See https://github.com/jomjol/AI-on-the-edge-device/actions/runs/12324445567/job/34402014328

**Comment by SybexX** [comment]
@caco3 versuche mal mit #include "../../include/basic_auth.h" statt #include "basic_auth.h" 
oder verschiebe basic_auth.h/basic_auth.cpp in den AI-on-the-edge-device-add-REST-auth\code\components\jomjol_wlan Ordner

**Comment by caco3** [comment]
Replaced by https://github.com/jomjol/AI-on-the-edge-device/pull/3436

File: code/platformio.ini (hunk 1 above)
Comment (caco3) [reply code/platformio.ini:60]: I don't see any relation

File: code/components/jomjol_wlan/read_wlanini.cpp (hunk 2 above)
Comment (caco3) [reply code/components/jomjol_wlan/read_wlanini.cpp:183]: I agree, the initial change does not make sense.

**Pull request closed without merging**